packages/components/src/color-picker/saturation.js

```diff
@@ -46,29 +46,21 @@ import Button from '../button';
 import KeyboardShortcuts from '../keyboard-shortcuts';
 import VisuallyHidden from '../visually-hidden';
 
-export class Saturation extends Component {
-	constructor( props ) {
-		super( props );
+export default function Saturation( { hsv, hsl, onChange = noop } ) {
+	const container = useRef();
+	const instanceId = useInstanceId( Saturation );
+	const [ mouseDown, setMouseDown ] = useState( false );
 
-		this.throttle = throttle( ( fn, data, e ) => {
-			fn( data, e );
-		}, 50 );
+	const pointerLocation = {
+		top: `${ -hsv.v + 100 }%`,
+		left: `${ hsv.s }%`,
+	};
 
-		this.container = createRef();
-		this.saturate = this.saturate.bind( this );
-		this.brighten = this.brighten.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.handleMouseUp = this.handleMouseUp.bind( this );
-	}
-
-	componentWillUnmount() {
-		this.throttle.cancel();
-		this.unbindEventListeners();
-	}
+	const throttle = _throttle( ( fn, data, e ) => {
+		fn( data, e );
+	}, 50 );
```

Comment:
Maybe you can create a new `useThrottle` hook in the `compose` package, similar to `useDebounce`?

Reply:
I added `useThrottle` hook daddou-ma/gutenberg@fce4404